components/oauth/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/token/OAuth2TokenEndpoint.java

```diff
@@ -197,27 +192,30 @@ private Response handleBasicAuthFailure() throws OAuthSystemException {
     }
 
     private void logAccessTokenRequest(HttpServletRequest request) {
-        log.debug("Received a request : " + request.getRequestURI());
-        // log the headers.
-        log.debug("----------logging request headers.----------");
-        Enumeration headerNames = request.getHeaderNames();
-        while (headerNames.hasMoreElements()) {
-            String headerName = (String) headerNames.nextElement();
-            Enumeration headers = request.getHeaders(headerName);
-            while (headers.hasMoreElements()) {
-                log.debug(headerName + " : " + headers.nextElement());
+
+        if (log.isDebugEnabled()){
+            log.debug("Received a request : " + request.getRequestURI());
+            // log the headers.
+            log.debug("----------logging request headers.----------");
+
+            Enumeration headerNames = request.getHeaderNames();
+            while (headerNames.hasMoreElements()) {
+                String headerName = (String) headerNames.nextElement();
+                Enumeration headers = request.getHeaders(headerName);
+                while (headers.hasMoreElements()) {
+                    log.debug(headerName + " : " + headers.nextElement());
+                }
             }
+            // log the parameters.
+            log.debug("----------logging request parameters.----------");
+            log.debug(OAuth.OAUTH_GRANT_TYPE + " - " + request.getParameter(OAuth.OAUTH_GRANT_TYPE));
+            log.debug(OAuth.OAUTH_CLIENT_ID + " - " + request.getParameter(OAuth.OAUTH_CLIENT_ID));
+            log.debug(OAuth.OAUTH_CODE + " - " + request.getParameter(OAuth.OAUTH_CODE));
+            log.debug(OAuth.OAUTH_REDIRECT_URI + " - " + request.getParameter(OAuth.OAUTH_REDIRECT_URI));
```

Comment:
Don't use multiple log.debug(), instead write to a StringBuilder and use one log.debug() to write to the log. Otherwise logs will get mix up with any other logs written concurrently.